src/mscorlib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -628,6 +628,49 @@ public bool Remove(TKey key)
             return false;
         }
 
+        public bool Remove(TKey key, out TValue value)
+        {
+            if (key == null)
+            {
+                value = default(TValue);
```

Comment:
What is the point of setting the value on exception thrown? It is not how similar method works. For example, `Dictionary.TryGetValue` won't set the value on exception.